### PR TITLE
style: add perf rules

### PR DESCRIPTION
**Pull request by Brooooooklyn**

### CODE REVIEW

**Comment by Brooooooklyn** [comment]
Current dependencies on/for this PR:
* `canary`
  * **PR #5413** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5413?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/toeverything/AFFiNE/5413?utm_source=stack-comment).

File: packages/frontend/core/src/components/affine/onboarding/articles/index.tsx [hunk 1]
```diff
@@ -244,8 +244,7 @@ const states: Partial<Record<ArticleId, EdgelessSwitchState>> = {
 
 export const articles: Record<ArticleId, ArticleOption> = ids.reduce(
   (acc, id) => {
-    return {
-      ...acc,
+    return Object.assign(acc, {
```
Comment (joooye34) [inline packages/frontend/core/src/components/affine/onboarding/articles/index.tsx:247]: This seems different code logic, change the origin object. Is it correct?

Comment (Brooooooklyn) [reply packages/frontend/core/src/components/affine/onboarding/articles/index.tsx:247]: the original object is created by the second param `{} as Record<ArticleId, ArticleOption>`, so it's totally okay here

Comment (pengx17) [reply packages/frontend/core/src/components/affine/onboarding/articles/index.tsx:247]: `{...acc, ...xxx}` is considered as bad practice now?

**Comment by Brooooooklyn** [comment]
Only if it was in `Array#reduce`, it's covered by the `oxlint -D perf`

File: packages/frontend/core/src/components/affine/onboarding/articles/index.tsx (hunk 1 above)
Comment (Brooooooklyn) [reply packages/frontend/core/src/components/affine/onboarding/articles/index.tsx:247]: Only if it was in `Array#reduce`, it's covered by the `oxlint -D perf`

Comment (pengx17) [reply packages/frontend/core/src/components/affine/onboarding/articles/index.tsx:247]: Cool! Now I get it,

**Review by pengx17 — APPROVED** [review]

**Pull request merged**